SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp

```diff
@@ -93,97 +93,149 @@ void UGlobalStateManager::ApplyAcceptingPlayersUpdate(bool bAcceptingPlayersUpda
-		{
-			check(Channel == nullptr || Channel->Actor == SingletonActor);
-			continue;
-		}
+		// No local actor has registered itself as replicatible on this worker
+		UE_LOG(LogGlobalStateManager, Log, TEXT("LinkExistingSingletonActor no actor registered"), *SingletonActorClass->GetName());
+		return;
+	}
 
-		SingletonActor->Role = ROLE_SimulatedProxy;
-		SingletonActor->RemoteRole = ROLE_Authority;
+	AActor*& SingletonActor = ActorChannelPair->Key;
+	USpatialActorChannel*& Channel = ActorChannelPair->Value;
 
-		// Add to entity registry
-		// This indirectly causes SetChannelActor to not create a new entity for this actor
-		NetDriver->GetEntityRegistry()->AddToRegistry(SingletonEntityId, SingletonActor);
+	if (Channel != nullptr)
+	{
+		// Channel has already been setup
+		UE_LOG(LogGlobalStateManager, Log, TEXT("UGlobalStateManager::LinkExistingSingletonActor channel already setup"), *SingletonActorClass->GetName());
+		return;
+	}
 
-		Channel->SetChannelActor(SingletonActor);
+	// If we have previously queued up data for this entity, apply it - JIRA: 734
 
-		// Since the entity already exists, we have to handle setting up the PackageMap properly for this Actor
-		FClassInfo* Info = NetDriver->TypebindingManager->FindClassInfoByClass(SingletonActor->GetClass());
-		check(Info);
+	// We're now ready to start replicating this actor, create a channel
+	USpatialNetConnection* Connection = Cast<USpatialNetConnection>(NetDriver->ClientConnections[0]);
+	Channel = Cast<USpatialActorChannel>(Connection->CreateChannel(CHTYPE_Actor, 1));
 
-		NetDriver->PackageMap->ResolveEntityActor(SingletonActor, SingletonEntityId, improbable::CreateOffsetMapFromActor(SingletonActor, Info));
+	SingletonActor->Role = ROLE_SimulatedProxy;
+	SingletonActor->RemoteRole = ROLE_Authority;
 
-		UE_LOG(LogGlobalStateManager, Log, TEXT("Linked Singleton Actor %s with id %d"), *SingletonActor->GetClass()->GetName(), SingletonEntityId);
-	}
+	// Add to entity registry
+	// This indirectly causes SetChannelActor to not create a new entity for this actor
+	NetDriver->GetEntityRegistry()->AddToRegistry(SingletonEntityId, SingletonActor);
+
+	Channel->SetChannelActor(SingletonActor);
+
+	// Since the entity already exists, we have to handle setting up the PackageMap properly for this Actor
+	FClassInfo* Info = NetDriver->TypebindingManager->FindClassInfoByClass(SingletonActor->GetClass());
+	check(Info);
+
+	NetDriver->PackageMap->ResolveEntityActor(SingletonActor, SingletonEntityId, improbable::CreateOffsetMapFromActor(SingletonActor, Info));
+
+	UE_LOG(LogGlobalStateManager, Log, TEXT("Linked Singleton Actor %s with id %d"), *SingletonActor->GetClass()->GetName(), SingletonEntityId);
 }
 
-void UGlobalStateManager::ExecuteInitialSingletonActorReplication()
+void UGlobalStateManager::LinkAllExistingSingletonActors()
 {
-	for (const auto& Pair : SingletonNameToEntityId)
+	// Client receive Singleton Actors via the normal Unreal replicated actor flow
```

Comment:
If a server isn't in charge of replicating the singletons, it would also hit this function?

Reply:
Yes, it gets called on servers when the singleton manager is updated.